[PyTorch][NCCL PG][Resubmit D67193887] Change getNCCLCommDumpMap to use new ncclCommDumpAll API

jiayulu opened the pull request:
Summary: see D67193887

Test Plan: https://www.internalfb.com/mlhub/pipelines/runs/mast/aps-dlrm_fsdp2_test-7711e496fc?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

Differential Revision: D74820576




cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta

Skylion007 commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -382,8 +391,11 @@ static std::
   for (auto& ncclComm : allNCCLComms) {
-    std::string ncclUniqueIDStr = buildNcclUniqueIdStr(ncclComm->getNcclId());
-    ncclDumpMap[ncclUniqueIDStr] = ncclComm->ncclCommDump();
+    std::stringstream ss;
+    ss << std::hex << ncclComm->getNcclUniqueHash();
+    std::string ncclUniqueHashStr = ss.str();
```
```suggestion
    std::string ncclUniqueHashStr = fmt::format("{:x}", ncclComm->getNcclUniqueHash());
```
More efficient and readable to use libfmt here?

Skylion007 commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -382,8 +391,11 @@ static std::
+    ss << std::hex << ncclComm->getNcclUniqueHash();
+    std::string ncclUniqueHashStr = ss.str();
+
+    ncclDumpMap[ncclUniqueHashStr] = ncclComm->ncclCommDump();
```
```suggestion
    ncclDumpMap[std::move(ncclUniqueHashStr)] =  ncclComm->ncclCommDump();
```

jiayulu committed:
[PyTorch][NCCL PG][Resubmit D67193887] Change getNCCLCommDumpMap to use new ncclCommDumpAll API (#153636)

Summary:

see D67193887

Test Plan: https://www.internalfb.com/mlhub/pipelines/runs/mast/aps-dlrm_fsdp2_test-7711e496fc?job_attempt=0&version=0&tab=execution_details&env=PRODUCTION

Differential Revision: D74820576
